lib/engine.js

```diff
@@ -872,19 +759,157 @@ module.exports = function (s, conf) {
+              executeOrder(order, type)
+              return syncBalance(function () {
+                cb(null, order)
+              })
+            }
+
+            s.api_order = api_order
+            if (api_order.filled_size) {
+              order.remaining_size = n(order.size).subtract(api_order.filled_size).format('0.00000000')
+            }
+          }
+          syncBalance(function () {
+            let on_hold
+            if (type === 'buy') on_hold = n(s.balance.currency).subtract(s.balance.currency_hold || 0).value() < n(order.price).multiply(order.remaining_size).value()
+            else on_hold = n(s.balance.asset).subtract(s.balance.asset_hold || 0).value() < n(order.remaining_size).value()
+
+            if (on_hold && s.balance.currency_hold > 0) {
+              // wait a bit for settlement
+              msg('funds on hold after cancel, waiting 5s')
+              queueEvent(function() { checkHold(do_reorder, cb) }, conf.wait_for_settlement)
+            }
+            else {
+              cb(null, do_reorder ? null : false)
+            }
+          })
+        })
+      }
+      checkHold(do_reorder, cb)
+    })
+  }
+  function checkOrder (order, type, cb) {
+    if (!s[type + '_order']) {
+      // signal switched, stop checking order
+      msg('signal switched during ' + type + ', aborting')
+      return cancelOrder(order, type, false, cb)
+    }
+    s.exchange.getOrder({order_id: order.order_id, product_id: s.product_id}, function (err, api_order) {
+      if (err) return cb(err)
+      s.api_order = api_order
+      order.status = api_order.status
+      if (api_order.reject_reason) order.reject_reason = api_order.reject_reason
+      if (api_order.status === 'done') {
+        order.time = new Date(api_order.done_at).getTime()
+        order.price = api_order.price || order.price // Use actual price if possible. In market order the actual price (api_order.price) could be very different from trade price
+        executeOrder(order, type)
+        return syncBalance(function () {
+          cb(null, order)
+        })
+      }
+      if (order.status === 'rejected' && (order.reject_reason === 'post only' || api_order.reject_reason === 'post only')) {
+        msg('post-only ' + type + ' failed, re-ordering')
+        return cb(null, null)
+      }
+      if (order.status === 'rejected' && order.reject_reason === 'balance') {
+        msg('not enough balance for ' + type + ', aborting')
+        return cb(null, null)
+      }
+      if (now() - order.local_time >= so.order_adjust_time) {
+        getQuote(function (err, quote) {
+          if (err) {
+            err.desc = 'could not execute ' + type + ': error fetching quote'
+            return cb(err)
+          }
+          let marked_price
+          if (type === 'buy') {
+            marked_price = nextBuyForQuote(s, quote)
+            if (so.exact_buy_orders && n(order.price).value() != marked_price) {
```

Comment:
What is `so.exact_buy|sell_orders` as it never seems to get populated?

Reply:
I wrote this as an additional strategy option. I've been using it in live for the last couple of weeks. The default `if (n(order.price).value() > marked_price)` is horrible for the strategy I'm running. It's very important to me that the price get adjusted in both directions so I add `--exact_buy_orders` to my command to solve that.

Reply:
> that the price get adjusted in both directions

Why no PR for this?

This might be useful to others as well and frankly, it makes little sense to have code in upstream that never gets reached.

Reply:
It's been a bit tricky trying to isolate all the things I'm working on at the same time. The only way I have to test this code is in live, because we don't have a good enough test framework in place already. So to pull these things out separately would be to break my bot, and then I don't have a way to test that what I'm doing is working or not. I agree with you in theory, but it's too hard in practice. Luckily it's written an as entirely optional feature that by default won't change anything. I'll go back and try to flesh out some documentation on this sort of thing later.

Reply:
Thats why i don't work on six different thinks at a time, well at least not in the same branch.

If you don't intend to do a PR for it simply stage the (for us) obsolete lines and include it once you are ready.

Git will take care of putting everything together again.

Reply:
If you say its ready why not simply include it in the PR then?

Update README and add it to trade/sim module.exports option so people actually know the options exists and what it does.

Will i usually discourage mixed commits its still better than adding something nobody knows about.

Reply:
That makes sense. I've added the documentation to those locations. I'll try to be better about splitting up work in the future. It was especially tricky this time.

Reply:
:+1: